assets/js/phoenix_live_view.js

```diff
@@ -2000,12 +2022,19 @@ export class View {
     return(
       this.liveSocket.wrapPush(() => {
         return this.channel.push(event, payload, PUSH_TIMEOUT).receive("ok", resp => {
-          if(resp.diff || ref !== null){ this.update(resp.diff || {}, payload.cid, ref) }
+          let hookReply = null
+          if(ref !== null){ DOM.undoRefs(ref, this.el) }
+          if(resp.diff){
+            this.applyDiff("update", resp.diff, ({diff, reply, events}) => {
+              hookReply = reply
+              this.update(diff, events, payload.cid)
+            })
```

Comment:
I just realized we have a slight event re-ordering here. Typically speaking, the reply callback in `pushEvent` is called *after* the join patch is applied. However, if we are queuing updates, we will mark the update as pending, and then call the `reply`, which will be invoked before the update.

We need to either also queue the reply or call the reply before the update. Thoughts?